### DISCUSSION TITLE
API documenation for otbr-web

### DISCUSSION THREAD

**Discussion by Vijay-Karthick** [openthread/openthread#7639] [Q&A]
Hi @jwhui 

Can you please point me to a link that shows API documentation for the backend APIs used in otbr-web application?

I'm specifically interested to know if there is any API to get information that can help me with thread network signal strength of nodes. Example when I issue [following command](https://github.com/openthread/openthread/blob/main/src/cli/README.md#neighbor-table), I can get this information. But I was looking for any API that may be available.
ot-ctl neighbor table

Kind regards,
Vijay

**Comment by wgtdkp** [comment]
Are you looking for a RESTful API? I admit there lacks documentation of the RESTful APIs :(

you may want to try the [/diagnostics](https://github.com/openthread/ot-br-posix/blob/96f176c9030b83392a009c8e5a983fb610672fc7/src/rest/resource.cpp#L38) API which returns all information defined [here](https://github.com/openthread/openthread/blob/9a2d84a4b78413e20f269df96a855e573e04a616/include/openthread/netdiag.h#L69-L84) as JSON payload for all Thread nodes. There may include information you are interested in, for example `OT_NETWORK_DIAGNOSTIC_TLV_CONNECTIVITY`.

**Reply by Vijay-Karthick** [reply]
Hi @wgtdkp 

Yes, I was looking for RESTful API.
Thanks for pointing me at the /diagnostics API. Though it provides interesting information, it doesn't contain RSSI information of the nodes. Do you know any other API that may provide this information?

**Reply by jwhui** [reply]
Thread Network Diagnostics does not directly provide RSSI information. You can learn more about Thread Network Diagnostics in [Thread 1.1.1 Section 10.11](https://www.threadgroup.org/support#specifications).

I'm not aware of any existing RESTful APIs that will provide what you want. However, it should be possible to add some that leverage the existing mechanisms used to retrieve neighbor and child tables. Contributions are always welcome :)
